uwblueprint/childrens-aid-society
# Fix heroku deployments on push to main

## Notion ticket link

[Configure GitHub actions Heroku deploy on merge to main](https://www.notion.so/uwblueprintexecs/Configure-GitHub-actions-Heroku-deploy-on-merge-to-main-c318c1a04b5c4b589762631a9f308c5b)



## Implementation description
* Allow backend code to be pushed to heroku when merged to main -> this will be our "staging" environment!
* Added DB credentials to Heroku secrets
* Note: we were having issues with installing `typed-ast` as it [doesn't support Python 3.8+](https://github.com/python/typed_ast#python-38) -- had to add a `runtime.txt` so we can pin the Heroku Python runtime version to `3.7`. `3.7.13` was chosen as it is the only version [supported by Heroku (stack v20)](https://devcenter.heroku.com/articles/python-support#specifying-a-python-version).



## Steps to test
1. Check that the [GitHub action for the previous commit](https://github.com/uwblueprint/childrens-aid-society/runs/7244684422?check_suite_focus=true) passed
2. Check that you can access [this link](https://childrens-aid-society.herokuapp.com/users) (it should say `"You are not authorized to make this request."`)



## What should reviewers focus on?
* Does it work for you?


## Checklist
